src/periodic.jl

```diff
@@ -43,11 +43,6 @@ function _apply(x, P::Periodic{T}; kwargs...) where T <: Period
     map(xi -> _periodic(P.f, xi, P.period, P.phase_shift), x)
 end
 
-function _apply!(x::AbstractArray{T}, P::Periodic; kwargs...) where T <: Real
```

Comment:
Not sure if this is informative benchmarking, but I was curious about the effect.

This branch:

```
julia> using BenchmarkTools, Transforms

julia> x = collect(1.:1_000_000.);

julia> p = Periodic(sin, 7);

julia> @benchmark Transforms.apply!(x, p)
BenchmarkTools.Trial: 
  memory estimate:  22.89 MiB
  allocs estimate:  9
  --------------
  minimum time:     6.602 ms (0.00% GC)
  median time:      10.553 ms (19.06% GC)
  mean time:        10.830 ms (33.86% GC)
  maximum time:     20.315 ms (63.28% GC)
  --------------
  samples:          462
  evals/sample:     1
```

`main` branch:

```
julia> @benchmark Transforms.apply!(x, p)
BenchmarkTools.Trial: 
  memory estimate:  22.89 MiB
  allocs estimate:  7
  --------------
  minimum time:     6.271 ms (0.00% GC)
  median time:      10.960 ms (17.62% GC)
  mean time:        10.793 ms (35.26% GC)
  maximum time:     21.596 ms (59.10% GC)
  --------------
  samples:          464
  evals/sample:     1
```

Why are there more allocations? What are approaches to reducing allocations? I'm not familiar with it.

Reply:
>Why are there more allocations? What are approaches to reducing allocations? I'm not familiar with it.

I'm not sure either, I gotta look into it.

One thing to note however, is that when using BenchmarkTools I think you have to interpolate the function args as this can affect the result https://github.com/JuliaCI/BenchmarkTools.jl#quick-start (note: this doesn't apply to `@time`)

Reply:
just to note that with `@views` this goes back to 7
```julia
julia> @benchmark Transforms.apply!($x, $p)
BenchmarkTools.Trial: 
  memory estimate:  22.89 MiB
  allocs estimate:  7
  --------------
  minimum time:     5.029 ms (0.00% GC)
  median time:      6.940 ms (16.92% GC)
  mean time:        6.427 ms (19.47% GC)
  maximum time:     9.906 ms (32.37% GC)
  --------------
  samples:          778
  evals/sample:     1
```